### ISSUE TITLE
Better UX for unsupported Python versions

### ISSUE THREAD

**Issue by prismv** [posit-dev/positron#5166]
#### Positron and OS details:

Positron Version: 2024.11.0 build 69
Code - OSS Version: 1.93.0
Commit: 4cde19a003d09cf75114ec8650020b132bbc5554
Date: 2024-10-21T02:40:21.993Z
Electron: 30.4.0
Chromium: 124.0.6367.243
Node.js: 20.15.1
V8: 12.4.254.20-electron.0
OS: Linux x64 6.6.58-1-lts

This issues exists also on Windows OS

#### Interpreter details:

3.13.0

## Describe the issue:

Code completion does not work in console or Jupyter notebook. Only "No suggestions." is shown on Tab key action

## Steps to reproduce the issue:

Miniforge3 was used: `mamba create -n py313 cpython=3.13`

Select this env, e.g. for console. Perform `import sys`, then `sys.` and then input Tab key to enforce code completion.

## Expected or desired behavior:

Completion with 3.12.7 kernel suggests already during typing some commands in pop-up.

## Were there any error messages in the UI, Output panel, or Developer Tools console?

workbench.desktop.main.js:130697
ERR NotImplementedError: Python version 3.13 is currently not supported.: Error: NotImplementedError: Python version 3.13 is currently not supported.

**Comment by juliasilge** [comment]
Related to https://github.com/posit-dev/positron/issues/5137

**Comment by prismv** [comment]
> I believe this is the same as https://github.com/posit-dev/positron/issues/5137. Do you mind taking a look and seeing if you have anything to add?

Yes. It is not obvious that some functionality is missing for some kernels. For unsupported (or even blocked?) versions there is no proper warning shown for the user.

**Comment by austin3dickey** [comment]
I am working on a PR that will
- show all interpreters that are found, instead of hiding interpreters that are too old for support
- clearly mark interpreters that are too old or new for support
- create a pop-up so users have to opt in to using an unsupported interpreter, which probably won't work

This should make the overall experience less confusing for users, especially when we stop supporting Python 3.8 in the near future.
